eng/pipelines/templates/jobs/ci.yml

```diff
@@ -226,18 +228,51 @@ jobs:
         parameters:
           BaselineFilePath: $(Build.sourcesdirectory)/eng/dotnet.gdnbaselines
           ServiceDirectory: ${{ parameters.ServiceDirectory }}
-
   - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
     parameters:
+      GenerateJobName: generate_target_service_test_matrix
       JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
       MatrixConfigs: ${{ parameters.MatrixConfigs }}
       MatrixFilters: ${{ parameters.MatrixFilters }}
       MatrixReplace: ${{ parameters.MatrixReplace }}
       CloudConfig:
-        Cloud: Public
+        Cloud: public
       AdditionalParameters:
         SDKType: ${{ parameters.SDKType }}
-        ServiceToTest: ${{ coalesce(parameters.ServiceToTest, parameters.ServiceDirectory) }}
         ServiceDirectory: ${{ parameters.ServiceDirectory }}
         TestSetupSteps: ${{ parameters.TestSetupSteps }}
-        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }} 
+  - ${{ if ne(parameters.TestDependsOnDependency, 'not-specified') }}:
+    - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+      parameters:
+        GenerateJobName: generate_target_dependencies_test_matrix 
+        JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+        MatrixConfigs: 
+          - Name: NET_ci_test_base
+            # The path here should refer to dependency.tests.yml parameter ProjectFilesOutputFolder 
+            Path: $(Build.ArtifactStagingDirectory)/platform-matrix.json
+            Selection: sparse
+            GenerateVMJobs: true
+        MatrixFilters: ${{ parameters.MatrixFilters }}
+        MatrixReplace: ${{ parameters.MatrixReplace }}
+        CloudConfig:
+          Cloud: public
+        SparseCheckoutPaths: 
+          - "/*"
+          - "!/sdk/*/**/SessionRecords"
+        PreGenerationSteps: 
+        - template: /eng/pipelines/templates/steps/dependency.tests.yml
```

Comment:
I know right now this just targets core, but is there a plan/need to support passing in matrix configs + additional matrix configs from the caller?

Reply:
Theoretically, we can use the same structure for other services as well, like storage, event-hubs